tests/testthat/test-independent-ahr_blinded.R

```diff
@@ -0,0 +1,161 @@
+test_that("ahr_blinded throws an error for non-numeric or negative hr", {
+  expect_error(
+    ahr_blinded(hr = c(1, -2, 3)),
+    "'hr' must be a vector of positive numbers."
+  )
+  expect_error(
+    ahr_blinded(hr = "abc"),
+    "'hr' must be a vector of positive numbers."
+  )
+})
+
+test_that("ahr_blinded throws an error when intervals and hr are not aligned", {
+  expect_error(
+    ahr_blinded(intervals = c(3, 6), hr = c(1)),
+    "The piecewise model specified 'hr' and 'intervals' differ in length."
+  )
+})
+
+test_that("ahr_blinded handles piecewise exponential model fitting and calculations correctly", {
+  surv <- survival::Surv(
+    time = simtrial::ex1_delayed_effect$month,
+    event = simtrial::ex1_delayed_effect$evntd
+  )
+  intervals <- c(3, 6, Inf)
+  hr <- c(1, 0.7, 0.5)
+  ratio <- 2
+
+  # Run the function
+  result <- ahr_blinded(surv = surv, intervals = intervals, hr = hr, ratio = ratio)
+
+  # Test 1: Correct fitting of survival data into piecewise exponential model
+  event <- simtrial::fit_pwexp(surv, intervals)[, 3]
+  expect_length(event, length(intervals))
+  expect_true(all(event >= 0))
+
+  # Test 2: Hazard ratio vector is correctly extended
+  nhr <- length(hr)
+  nx <- length(event)
+  if (length(hr) < length(event)) {
+    hr <- c(hr, rep(hr[nhr], nx - nhr))
+  }
+  expect_equal(length(hr), length(event))
+  expect_equal(hr, c(1, 0.7, 0.5)) # Expected extended hr vector
+
+  # Test 3: Blinded AHR (theta) is computed correctly
+  theta <- -sum(log(hr[1:nx]) * event) / sum(event)
```

Comment:
It looks like the calculation of `theta` is exactly the same as what we have in https://github.com/Merck/gsDesign2/blob/main/R/ahr_blinded.R#L109. So, I am wondering if it is a validated independent testing or not.